src/main/java/org/opensearch/knn/index/KNNSettings.java

```diff
@@ -397,6 +410,14 @@ public static Integer getFilteredExactSearchThreshold(final String indexName) {
             .getAsInt(ADVANCED_FILTERED_EXACT_SEARCH_THRESHOLD, ADVANCED_FILTERED_EXACT_SEARCH_THRESHOLD_DEFAULT_VALUE);
     }
 
+    public static boolean isKNNSyntheticEnabled(final String indexName) {
```

Comment:
Lets add java doc on all the public functions.

Reply:
FIXED at https://github.com/opensearch-project/k-NN/pull/1571/commits/f4a69ca2f04833f238a54d76e75a9ed4f3db0dcb